### PR TITLE
HOSTEDCP-2057: [CPO Refactor] Refactor ETCD

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/v2/etcd/component.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+func (e *etcd) IsRequestServing() bool {
+	return false
+}
+
+// MultiZoneSpread implements controlplanecomponent.ComponentOptions.
+func (e *etcd) MultiZoneSpread() bool {
+	return true
+}
+
+// NeedsManagementKASAccess implements controlplanecomponent.ComponentOptions.
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/v2/etcd/component.go:27]: It'd be nice if we document some of these as we go through.
E.g. // etcd needs management kas access for some controllers to implement leader election, e.g. etcd-defrag-controller

File: support/controlplane-component/status.go [hunk 2]
```diff
@@ -21,14 +21,20 @@ import (
...
+	kubeAPIServerComponentName = "kube-apiserver"
+	etcdComponentName          = "etcd"
 )
 
 func (c *controlPlaneWorkload) checkDependencies(cpContext ControlPlaneContext) ([]string, error) {
 	unavailableDependencies := sets.New(c.dependencies...)
-	// always add kube-apiserver as a dependency.
-	// TODO: don't add for etcd component once refactored.
-	unavailableDependencies.Insert(KubeAPIServerComponentName)
+	// always add kube-apiserver as a dependency, except for etcd.
```
Comment (enxebre) [inline support/controlplane-component/status.go:30]: can we please unit test this func?

Comment (muraee) [reply support/controlplane-component/status.go:30]: created a ticket https://issues.redhat.com/browse/HOSTEDCP-2068

File: control-plane-operator/controllers/hostedcontrolplane/v2/etcd/component.go (hunk 1 above)
Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/v2/etcd/component.go:27]: done

**Pull request merged**